front/src/components/AdminPanel/Panel/CompetitionDetails/EditDetails/EditCompetitionDetailsForm.js

```diff
@@ -1,26 +1,24 @@
 import React, { useState } from "react";
-import { useHistory } from "react-router-dom";
+import { useHistory, useParams } from "react-router-dom";
 import Button from "../../../../UI/Button/Button";
 import competitionDetaildata from "../../../competitionsData.json";
-
-function revertDateFormat(dateStr) {
-    const arr = dateStr.split("-");
-    const res = arr[2] + "-" + arr[1] + "-" + arr[0];
-    return res;
-}
+import { handleSubmit } from "./handleSubmit.js";
 
 const CompetitionForm = (props) => {
     const history = useHistory();
+    const params = useParams();
+    const _id = params.id;
 
-    const _id = props.id;
-    const _name = competitionDetaildata[_id - 1]["name"];
-    const _start_date = revertDateFormat(
-        competitionDetaildata[_id - 1]["date_start"]
-    );
-    const _end_date = revertDateFormat(
-        competitionDetaildata[_id - 1]["date_end"]
-    );
-    const _location = competitionDetaildata[_id - 1]["location"];
+    let desiredCompetition = {};
+    Object.keys(competitionDetaildata).forEach((oneComp) => {
+        if (competitionDetaildata[oneComp].id === parseInt(_id))
+            desiredCompetition = competitionDetaildata[oneComp];
+    });
```

Comment:
Same lines as in CompetitionDetails.js, use one-liner I suggested there or extract to separate function.